src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -529,6 +523,7 @@ static URLConnection openURLConnection(URL url, String credentials, String proxy
             String encoding = Base64.getEncoder().encodeToString(proxyCredentials.getBytes(StandardCharsets.UTF_8));
             con.setRequestProperty("Proxy-Authorization", "Basic " + encoding);
         }
+        con.setRequestProperty(JnlpConnectionState.CLIENT_NAME_KEY, agentName);
```

Comment:
There are duplicate copies of this code at

```
src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java
526:            con.setRequestProperty("Authorization", "Basic " + encoding);
530:            con.setRequestProperty("Proxy-Authorization", "Basic " + encoding);

src/main/java/hudson/remoting/Util.java
131:            con.setRequestProperty("Authorization", "Basic " + encoding);
135:            con.setRequestProperty("Proxy-Authorization", "Basic " + encoding);

src/main/java/hudson/remoting/Launcher.java
500:                        http.setRequestProperty("Authorization", "Basic " + encoding);
504:                        http.setRequestProperty("Proxy-Authorization", "Basic " + encoding);
```

raiseing the question: why change this copy but not the others?

Reply:
https://github.com/jenkinsci/remoting/blob/5491eb1a523ec1d0fa98a2c2e763a38f5acd8820/src/main/java/hudson/remoting/Util.java#L103-L158 is used only from https://github.com/jenkinsci/remoting/blob/5491eb1a523ec1d0fa98a2c2e763a38f5acd8820/src/main/java/hudson/remoting/Launcher.java#L480-L507 and this is only used for the `*.jnlp` retrieval from the `java -jar` entrypoint in which the agent name is already encoded in the URL.

Reply:
Indeed there are three copies of the code to define `Authorization` and `Proxy-Authorization`, and two of the code to use `Proxy` (some of this was noticed in #243), as well as some unused overloads and other dead code, and it would be nice to clean this sort of thing up. That does not have any particular bearing on this PR since the `Node-Name` header is only useful in the `/tcpSlaveAgentListener/` request made from `JnlpAgentEndpointResolver`.

Reply:
> That does not have any particular bearing on this PR

It does have particular bearing on this PR, since this PR is making the existing problem worse by increasing the divergence between the three copies, thereby increasing the maintenance burden on the Jenkins community.

Reply:
Well at any rate #687 cleans up code without any behavioral change.